src/authentication/authentication-hapi-plugin.ts

```diff
@@ -385,42 +390,45 @@ class AuthenticationHapiPlugin extends HapiPlugin {
     }
   }
 
-  private userHasAccessToTeam(userNameOrRequest: Hapi.Request | string) {
-    return async (teamId: number, shouldThrowOnFalse = false) => {
+  private userHasAccessToTeamDecorator(request: Hapi.Request) {
+    return async(teamId: number) => {
       try {
-        // the username field is set by the 'validateFuncFactory'
-        const userName = typeof userNameOrRequest === 'string' ?
-          userNameOrRequest :
-          userNameOrRequest.auth.credentials.username;
-        return (await this.gitlab.getGroup(teamId, userName)).id === teamId;
+        return this.userHasAccessToTeam(this.getUserName(request), teamId);
       } catch (error) {
-        this.logger.warn(`User not authorized to access team ${teamId}`, error);
-        if (shouldThrowOnFalse) {
-          throw error;
-        }
+        this.logger.warn(`Can't check authorization since user is not authenticated`, error);
         return false;
       }
     };
   }
 
-  private userHasAccessToProject(userNameOrRequest: Hapi.Request | string) {
-    return async (projectId: number, shouldThrowOnFalse = false) => {
+  private userHasAccessToProjectDecorator(request: Hapi.Request) {
+    return async (projectId: number) => {
       try {
-        // the username field is set by the 'validateFuncFactory'
-        const userName = typeof userNameOrRequest === 'string' ?
-          userNameOrRequest :
-          userNameOrRequest.auth.credentials.username;
-        return (await this.gitlab.getProject(projectId, userName)).id === projectId;
+        return this.userHasAccessToProject(this.getUserName(request), projectId);
       } catch (error) {
-        this.logger.warn(`User not authorized to access project ${projectId}`, error);
-        if (shouldThrowOnFalse) {
-          throw error;
-        }
+        this.logger.warn(`Can't check authorization since user is not authenticated`, error);
         return false;
       }
     };
   }
 
+  public async userHasAccessToTeam(userName: string, teamId: number) {
+    try {
+      return (await this.gitlab.getGroup(teamId, userName)) !== undefined;
```

Comment:
I feel like I'm missing something.

What I'm trying to undertand is, why not write:
```typescript
await this.gitlab.getGroup(teamId, userName));
return true;
```

What is the purpose of the check? Note that current implementation for `userHasAccessToTeam` would return `true` for `getGroup` returning `null` or an empty string from.

If the point is to make a sanity check for JSON that GitLab returns, I would
I would do such a check in the `getGroup`, throwing if the payload does not pass the sanity check.